parasoft-findings-teamcity-agent/src/main/resources/com/parasoft/findings/teamcity/agent/sa-pmd.xsl

```diff
@@ -121,4 +131,19 @@
         </xsl:if>
     </xsl:template>
 
+    <xsl:template name="substring-after-last">
+        <xsl:param name="string" />
+        <xsl:param name="delimiter" />
+        <xsl:if test="contains($string, $delimiter)">
```

Comment:
What I said we should try to use <xsl:if> more, I mean we should try to avoid using <xsl:choose> if <xsl:if> is more suitable because <xsl:if> is more efficient. 
But for case here which has a "either-or" situation, we should use <xsl:choose> for better readability.